init-native-tools failing builds

https://github.com/dotnet/arcade/pull/3359 seems to be failing builds

See https://dev.azure.com/dnceng/internal/_build/results?buildId=276755

/cc @jonfortescue, @alexperovich, @chcosta 

```
##[section]Starting: Windows Build / Publish
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.151.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
eng\common\cibuild.cmd -configuration Debug -prepareMachine    /p:Platform=x86
========================== Starting Command Output ===========================
##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "F:\workspace\_work\_temp\21702468-df65-44b4-9a63-6916b7027f79.cmd""
at InitializeNativeTools, F:\workspace\_work\1\s\eng\common\tools.ps1: line 422
at <ScriptBlock>, F:\workspace\_work\1\s\eng\common\Build.ps1: line 130
at <ScriptBlock>, <No file>: line 1
##[error](NETCORE_ENGINEERING_TELEMETRY=InitializeToolset) At F:\workspace\_work\1\s\eng\common\init-tools-native.ps1:101 char:96
+ ... tallerPath $($LocalInstallerArguments.Keys.ForEach({"-$_ `"$($LocalIn ...
+                                                                ~~
Unexpected token '$(' in expression or statement.

At F:\workspace\_work\1\s\eng\common\init-tools-native.ps1:101 char:126
+ ... uments.Keys.ForEach({"-$_ `"$($LocalInstallerArguments.$_)`""}) -join ...
+                                                               ~~~
Unexpected token '`""' in expression or statement.
Killing running build processes...
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Windows Build / Publish
```

--- a/eng/common/init-tools-native.ps1
+++ b/eng/common/init-tools-native.ps1
@@ -98,7 +98,7 @@ try {
       }
 
       Write-Verbose "Installing $ToolName version $ToolVersion"
-      Write-Verbose "Executing '$InstallerPath $($LocalInstallerArguments.Keys.ForEach({"-$_ `"$($LocalInstallerArguments.$_)`""}) -join ' ')'"
+      Write-Verbose "Executing '$InstallerPath $LocalInstallerArguments'"
       & $InstallerPath @LocalInstallerArguments
       if ($LASTEXITCODE -Ne "0") {
         $errMsg = "$ToolName installation failed"

